packages/x-data-grid/src/utils/styled.ts

```diff
@@ -0,0 +1,6 @@
+import { createStyled } from '@mui/system';
+
+// Hardcoded theme id to avoid importing from `@mui/material`
+const MATERIAL_THEME_ID = '$$material';
+
+export const styled = createStyled({ themeId: MATERIAL_THEME_ID });
```

Comment:
```suggestion
export const styled = createStyled<Theme>({ themeId: MATERIAL_THEME_ID });
```

The `Theme` should be the Material UI Theme + other properties that data grid wants to extend.